### PR TITLE
Introduced two version support feature 

### PR DESCRIPTION
**Pull request by hlts2**
Now that Vald can support multiple versions, clients should also be able to support multiple versions.
In this PR, I implemented a workflow to support multiple versions.
The main logic is implemented as a common workflow in [vald-client-ci](https://github.com/vdaas/vald-client-ci), so I implemented a call to it.

## Summary by CodeRabbit

- **New Features**
	- Introduced workflows for end-to-end testing, release management, and repository synchronization using GitHub Actions.
	- Updated the Makefile to support new Python version and testing configurations.
- **Chores**
	- Updated `.gitignore` to exclude a specific JSON file.
- **Documentation**
	- Added licensing and workflow details in GitHub Actions workflow files.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -1,102 +1,32 @@
...
-          python3 setup.py sdist
-          python3 setup.py bdist_wheel
-      - name: publish
-        uses: pypa/gh-action-pypi-publish@master
-        with:
-          user: ${{ secrets.PIP_USERNAME }}
-          password: ${{ secrets.PIP_TOKEN }}
+    # TODO: change to the following before main merge.
+    # uses: vdaas/vald-client-ci/.github/workflows/_release.yaml@main
+    uses: vdaas/vald-client-ci/.github/workflows/_release.yaml@feat/add-common-workflows
```
Comment (hlts2) [inline .github/workflows/release.yml]:
```suggestion
    # uses: vdaas/vald-client-ci/.github/workflows/release.yaml@main
    uses: vdaas/vald-client-ci/.github/workflows/release.yaml@feat/add-common-workflows
```

File: Makefile [hunk 2]
```diff
@@ -121,48 +134,68 @@ vald/sha/print:
...
+.PHONY: ci/package/prepare
+## prepare package to publish
+ci/package/prepare:
+	python3 setup.py sdist
+	python3 setup.py bdist_wheel
+
+.PHONY: ci/package/publish
+## publich packages
+ci/package/publish:
+	@echo "Nothing do be done"
```
Comment (hlts2) [inline Makefile]:
```suggestion
.PHONY: ci/deps/update
## update deps for CI environment
ci/deps/update:
	@echo "Nothing to do"

.PHONY: ci/package/prepare
## prepare package to publish
ci/package/prepare:
	python3 setup.py sdist
	python3 setup.py bdist_wheel

.PHONY: ci/package/publish
## publich packages
ci/package/publish:
	@echo "Nothing to do"
```

**Comment by hlts2** [comment]
@coderabbitai review

File: Makefile [hunk 3]
```diff
@@ -121,48 +134,68 @@ vald/sha/print:
...
-	    echo "$${new_version}" > VALD_CLIENT_PYTHON_VERSION)
+	    echo "VALD_VERSION: $${vald_version}"; \
+	    echo "$${vald_version}" > VALD_CLIENT_PYTHON_VERSION)
 	sed -i -e "s/^version = .*\$$/version = `cat VALD_CLIENT_PYTHON_VERSION`/" setup.cfg
 
-.PHONY: proto/deps
+.PHONY: test
+## Execute test
+test: $(TEST_DATASET_PATH)
+	python src/test.py
```
Comment (datelier) [inline Makefile]:
```suggestion
	$(PYTHON) src/test.py
```

File: Makefile [hunk 4]
```diff
@@ -121,48 +134,68 @@ vald/sha/print:
...
+
+.PHONY: ci/deps/install
+## install deps for CI environment
+ci/deps/install: proto/deps/install
+	sudo apt-get update -y && sudo apt-get install -y \
+		python3-setuptools \
+		libprotobuf-dev \
+		libprotoc-dev \
+		protobuf-compiler
+	pip3 install grpcio-tools
```
Comment (datelier) [inline Makefile]: How do you feel about this suggestion?

```suggestion
PIP = pip3

.PHONY: ci/deps/install
## install deps for CI environment
ci/deps/install: proto/deps/install
	sudo apt-get update -y && sudo apt-get install -y \
		python3-setuptools \
		libprotobuf-dev \
		libprotoc-dev \
		protobuf-compiler
	$(PIP) install grpcio-tools
```

File: Makefile (hunk 3 above)
Comment (hlts2) [reply Makefile]: Thank you for catching it 🙏 I apply your suggestion!

File: Makefile (hunk 4 above)
Comment (hlts2) [reply Makefile]: Thank you for your suggestion 🙏  It looks good, so I apply your suggestion!

**Review by datelier — APPROVED** [review]

**Pull request merged**